src/panoptes/pocs/camera/camera.py

```diff
@@ -921,20 +927,53 @@ def _setup_observation(self, observation, headers, filename, **kwargs):
             metadata['filter_request'] = observation.filter_name
 
         if headers is not None:
+            self.logger.trace(f'Updating {file_path} metadata with provided headers')
             metadata.update(headers)
 
-        self.logger.debug(
-            f'Observation setup: exptime={exptime} file_path={file_path} image_id={image_id} '
-            f'metadata={metadata}')
+        self.logger.debug(f'Observation setup: {exptime=} {file_path=} {image_id=} {metadata=}')
+
         return exptime, file_path, image_id, metadata
 
-    def _process_fits(self, file_path, info):
+    def _process_fits(self, file_path, metadata):
         """
-        Add FITS headers from info the same as images.cr2_to_fits()
+        Add FITS headers from metadata the same as images.cr2_to_fits()
         """
-        self.logger.debug(f"Updating FITS headers: {file_path}")
-        fits_utils.update_observation_headers(file_path, info)
-        self.logger.debug(f"Finished FITS headers: {file_path}")
+        # TODO (wtgee) I don't like this one bit.
+        fields = {
+            'image_id': {'keyword': 'IMAGEID'},
+            'sequence_id': {'keyword': 'SEQID'},
+            'field_name': {'keyword': 'FIELD'},
+            'ra_mnt': {'keyword': 'RA-MNT', 'comment': 'Degrees'},
+            'ha_mnt': {'keyword': 'HA-MNT', 'comment': 'Degrees'},
+            'dec_mnt': {'keyword': 'DEC-MNT', 'comment': 'Degrees'},
+            'equinox': {'keyword': 'EQUINOX', 'default': 2000.},
+            'airmass': {'keyword': 'AIRMASS', 'comment': 'Sec(z)'},
+            'filter': {'keyword': 'FILTER'},
+            'latitude': {'keyword': 'LAT-OBS', 'comment': 'Degrees'},
+            'longitude': {'keyword': 'LONG-OBS', 'comment': 'Degrees'},
+            'elevation': {'keyword': 'ELEV-OBS', 'comment': 'Meters'},
+            'moon_separation': {'keyword': 'MOONSEP', 'comment': 'Degrees'},
+            'moon_fraction': {'keyword': 'MOONFRAC'},
+            'creator': {'keyword': 'CREATOR', 'comment': 'POCS Software version'},
+            'camera_uid': {'keyword': 'INSTRUME', 'comment': 'Camera ID'},
+            'observer': {'keyword': 'OBSERVER', 'comment': 'PANOPTES Unit ID'},
+            'origin': {'keyword': 'ORIGIN'},
+            'tracking_rate_ra': {'keyword': 'RA-RATE', 'comment': 'RA Tracking Rate'},
+        }
+
+        self.logger.debug(f"Updating FITS headers: {file_path} with {metadata=}")
+        with fits.open(file_path, 'update') as f:
+            hdu = f[0]
+            for metadata_key, field_info in fields.items():
```

Comment:
So only certain fields will be written to the file. This seems like fine default behaviour, but if I passed some non-standard headers to `take_observation`, I would expect them to be written to file, rather than to be ignored without warning.

Reply:
Agreed, but outside the scope of this PR unfortunately.

Reply:
The loop could be rewritten as:

```py
            for metadata_key, metadata_value in metadata.items():
                try:
                    field_info = fields[metadata_key]
                    # for metadata_key, field_info in fields.items():
                    fits_key = field_info['keyword']
                    fits_comment = field_info.get('comment', '')
                    # Get the value from either the metadata, the default, or use blank string.
                    fits_value = metadata.get(metadata_key, field_info.get('default', ''))

                    self.logger.trace(f'Setting {fits_key=} = {fits_value=} {fits_comment=}')
                except KeyError:
                    self.logger.trace(f'No mapping found for metadata {metadata_key}'
                                      f', will attempt to add as-is')
                    fits_key = metadata_key
                    fits_comment = ''
                    fits_value = metadata_value

                hdu.header.set(fits_key, fits_value, fits_comment)
```

That allows random things to go into the fits headers although does no check for a valid fits header. Astropy will automatically create it with a `HIERARCH` card, which I don't know much about.